extensions-support/httpclient5/deployment/src/main/java/org/apache/camel/quarkus/support/httpclient5/deployment/HttpClient5Processor.java

```diff
@@ -34,4 +46,54 @@ NativeImageResourceBuildItem suffixListResource() {
     void runtimeInitializedClasses(BuildProducer<RuntimeInitializedClassBuildItem> runtimeInitializedClasses) {
         runtimeInitializedClasses.produce(new RuntimeInitializedClassBuildItem(NTLM_ENGINE_IMPL));
     }
+
+    @BuildStep(onlyIf = { NativeOrNativeSourcesBuild.class, BrotliAbsentBooleanSupplier.class })
+    void generateBrotliInputStreamClass(BuildProducer<GeneratedClassBuildItem> generatedClass) {
```

Comment:
@jamesnetherton  Good work - the only thing I concern is that if there is something change with `brotli` in the future, this class name maybe change  or whatever. We have to rewrite it?